pkg/controller/sparkapplication/controller.go

```diff
@@ -72,9 +70,8 @@ type Controller struct {
 	metrics           *sparkAppMetrics
 	applicationLister crdlisters.SparkApplicationLister
 	podLister         v1.PodLister
-	jobLister         batchv1.JobLister
 	ingressURLFormat  string
-	subJobManager     *submissionJobManager
+	subJobManager     submissionJobManagerIface
```

Comment:
nit: let's name this interface `submissionJobManager` and the real implementation `realSubmissionJobManager` and the fake one for testing `fakeSubmissionJobManager`.

Reply:
done